src/base/rss/rss_feed.cpp

```diff
@@ -227,24 +227,71 @@ void Feed::handleParsingFinished(const RSS::Private::ParsingResult &result)
 
         m_lastBuildDate = result.lastBuildDate;
 
-        int newArticlesCount = 0;
         const QDateTime now {QDateTime::currentDateTime()};
-        for (QVariantHash varHash : result.articles) {
+        QVector<QVariantHash> newArticles;
+        newArticles.reserve(result.articles.size());
+        for (QVariantHash article : result.articles) {
+            QVariant &torrentURL = article[Article::KeyTorrentURL];
+            if (torrentURL.toString().isEmpty())
+                torrentURL = article[Article::KeyLink];
+
+            // if item does not have a guid, fall back to some other identifier
+            QVariant &id = article[Article::KeyId];
+            if (id.toString().isEmpty())
+                id = article.value(Article::KeyTorrentURL);
+            if (id.toString().isEmpty())
+                id = article.value(Article::KeyTitle);
+
+            // NOTE: Maybe make parser to be more strict?
+            if (id.toString().isEmpty())
+                continue;
+
+            if (articleByGUID(id.toString()))
+                continue;
+
+            QVariant &articleDate = article[Article::KeyDate];
             // if article has no publication date we use feed update time as a fallback
-            QVariant &articleDate = varHash[Article::KeyDate];
             if (!articleDate.toDateTime().isValid())
                 articleDate = now;
 
-            try {
-                auto article = new Article(this, varHash);
-                if (addArticle(article))
-                    ++newArticlesCount;
-                else
-                    delete article;
-            }
-            catch (const std::runtime_error&) {}
+            newArticles.append(article);
         }
 
+        using ArticleSortAdaptor = QPair<QDateTime, const QVariantHash *>;
+        std::vector<ArticleSortAdaptor> sortData;
+        const QList<Article *> existingArticles = this->articles();
```

Comment:
Nitpick, I some how prefer grouping it in this order:
```c++
const QList<Article *> existingArticles = this->articles();
std::vector<ArticleSortAdaptor> sortData;
sortData.reserve(existingArticles.size() + newArticles.size());
```

Reply:
Ok.